tools/cjxl_ng_main.cc

```diff
@@ -0,0 +1,597 @@
+    FLAGS_helpshort = true;
+    gflags::HandleCommandLineHelpFlags();
+    return EXIT_FAILURE;
+  }
+  const char* filename_in = argv[1];
+  const char* filename_out = argv[2];
+
+  size_t num_worker_threads = JxlThreadParallelRunnerDefaultNumWorkerThreads();
+  {
+    int64_t flag_num_worker_threads = FLAGS_num_threads;
+    if (flag_num_worker_threads != 0) {
+      num_worker_threads = flag_num_worker_threads;
+    }
+  }
+  // ManagedJxlEncoder managed_jxl_encoder =
+  //  ManagedJxlEncoder(num_worker_threads);
+  // managed_jxl_encoder.encoder_;
+  auto enc = JxlEncoderMake(/*memory_manager=*/nullptr);
+  auto runner = JxlThreadParallelRunnerMake(
+      /*memory_manager=*/nullptr, num_worker_threads);
+  if (JXL_ENC_SUCCESS != JxlEncoderSetParallelRunner(enc.get(),
+                                                     JxlThreadParallelRunner,
+                                                     runner.get())) {
+    fprintf(stderr, "JxlEncoderSetParallelRunner failed\n");
+    return EXIT_FAILURE;
+  }
+  JxlEncoder* jxl_encoder = enc.get();
+
+  const int32_t store_jpeg_metadata = FLAGS_store_jpeg_metadata;
+  if (!(-1 <= store_jpeg_metadata && store_jpeg_metadata <= 1)) {
+    std::cerr
+        << "Invalid --store_jpeg_metadata. Valid values are {-1, 0, 1}.\n";
+    return EXIT_FAILURE;
+  }
+  if (store_jpeg_metadata != -1) {
+    if (JXL_ENC_SUCCESS !=
+        JxlEncoderStoreJPEGMetadata(jxl_encoder, store_jpeg_metadata != 0)) {
+      std::cerr << "JxlEncoderStoreJPEGMetadata failed\n";
+      return EXIT_FAILURE;
+    }
+  }
+
+  JxlEncoderFrameSettings* jxl_encoder_frame_settings =
+      JxlEncoderFrameSettingsCreate(enc.get(), nullptr);
+
+  {  // Processing tuning flags.
+    bool use_container = FLAGS_container;
+    // TODO(tfish): Set use_container according to need of encoded data.
+    // This will likely require moving this piece out of flags-processing.
+    if (FLAGS_strip) {
+      use_container = false;
+    }
+    JxlEncoderUseContainer(jxl_encoder, use_container);
+
+    ProcessTristateFlag("modular", FLAGS_modular, jxl_encoder_frame_settings,
+                        JXL_ENC_FRAME_SETTING_MODULAR);
+    ProcessTristateFlag("keep_invisible", FLAGS_keep_invisible,
+                        jxl_encoder_frame_settings,
+                        JXL_ENC_FRAME_SETTING_KEEP_INVISIBLE);
+    ProcessTristateFlag("dots", FLAGS_dots, jxl_encoder_frame_settings,
+                        JXL_ENC_FRAME_SETTING_DOTS);
+    ProcessTristateFlag("patches", FLAGS_patches, jxl_encoder_frame_settings,
+                        JXL_ENC_FRAME_SETTING_PATCHES);
+    ProcessTristateFlag("gaborish", FLAGS_gaborish, jxl_encoder_frame_settings,
+                        JXL_ENC_FRAME_SETTING_GABORISH);
+    ProcessTristateFlag("group_order", FLAGS_group_order,
+                        jxl_encoder_frame_settings,
+                        JXL_ENC_FRAME_SETTING_GROUP_ORDER);
+
+    const int32_t flag_effort = FLAGS_effort;
```

Comment:
Maybe a validator with a (template?) parameter for the list of valid values would overall reduce boilerplate, given the number of such checks we have.

Reply:
I think it would be a good idea: added TODO for that!